assets/js/panel.js

```diff
@@ -0,0 +1,197 @@
+    }
+
+    // enable the info button;
+    // provides a way to get back to the intro on mobile
+    this.infoButton.disabled = false;
+  }
+
+  showIntro() {
+    // dispatch panel-close event;
+    // handler in leaf code will close leaf details
+    // (showing the intro implies closing leaf details if a leaf is selected)
+    this.el.dispatchEvent(PanelCloseEvent);
+    // open the panel without showing leaf details section
+    this.open(false);
+  }
+
+  closeIntro() {
+    // close the panel without closing leaf details
+    this.close(false);
+  }
+
+  announce(content) {
+    // update the contents of the aria live region
+    // to make polite announcements for screen reader users
+    this.status.textContent = content;
+  }
+
+  loadURL(url, callback) {
+    // load specified url; display article contents in the panel
+    // takes an optional callback; if defined, will be applied to
+    // the url contents before inserting
+
+    // if the requested url is already showing, do nothing
+    if (this.el.dataset.showing && this.el.dataset.showing == url) {
+      return;
+    }
+    // If you need to test load failure, uncomment this
+    // if (Math.random() < 0.5) {
+    //   url = url + "xxx";
+    // }
+    // console.log("fetching:", url);
+
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          return Promise.reject(response);
+        }
+        return response;
+      })
+      .then((response) => response.text())
+      .then((html) => {
+        let parser = new DOMParser();
+        const doc = parser.parseFromString(html, "text/html");
+        // Get the article content and insert into panel
+        const article = doc.querySelector("article");
+        if (callback != undefined) {
+          callback(article);
+        }
+        this.el.querySelector("article").replaceWith(article);
+        // store loaded url in data attribute for reload check
+        this.el.dataset.showing = url;
+      })
+      .catch((response) => {
+        // if the request failed, display error article
+        let errorArticle = document.querySelector("#loaderror").cloneNode(true);
+        this.el.querySelector("article").replaceWith(errorArticle);
+      });
+
+    // scroll to the top, in case previous leaf was scrolled
+    this.el.scrollTop = 0;
+
+    // ensure the panel is open with details shown
+    this.open();
+  }
+
+  bindHandlers() {
+    // bind click event handler for panel close button
+    document
+      .querySelector("aside .close")
+      .addEventListener("click", this.close.bind(this));
+
+    this.infoButton.addEventListener("click", this.showIntro.bind(this));
+
+    // bind a delegated click handler to handle scrolling on footnote links
+    const asideContainer = document.querySelector("aside");
+    asideContainer.addEventListener(
+      "click",
+      this.handleFootnoteLinkClick.bind(this)
+    );
+  }
+
+  handleFootnoteLinkClick(event) {
```

Comment:
I couldn't find an example article with a footnote to test this, but it seems like a clever way to standardize relative scroll behavior across browsers